patch/ethersCompat.d.ts

```diff
@@ -0,0 +1,12 @@
+// This file is used to patch a compatibility issue with siwe and
+// ethers v5.
+
+import { ethers } from 'ethers';
+type ProviderV5 = ethers.providers.Provider;
+// @ts-expect-error -- v6 compatibility hack
+type ProviderV6 = ethers.Provider;
+export type Provider = ProviderV6 extends undefined ? ProviderV5 : ProviderV6;
+export declare const verifyMessage: any;
+export declare const hashMessage: any;
+export declare const getAddress: any;
+export { };
```

Comment:
why are we using `ethers` v6 and not `5.7`

Reply:
is this for HD key derivation support?

Reply:
Yes, and we are using either v5. This is to work around a bug in siwe that tries to handle both versions, but just ends up just causing TS to error out when using v5.7.